src/System.Numerics.Matrices/src/System/Numerics/MatrixTemplate.tt

```diff
@@ -0,0 +1,611 @@
+        /// </summary>
+<#
+        for (int y = 0; y < rowCount; y++)
+        {
+            for (int x = 0; x < colCount; x++)
+            {
+                #>        /// <param name="m<#= x + 1 #><#= y + 1 #>">The column <#= x + 1 #>, row <#= y + 1 #> value</param><#
+                #><#= Environment.NewLine #><#
+            }
+        }
+#>
+        public <#= structName #>(<#
+            for (int y = 0; y < rowCount; y++)
+            {
+                for (int x = 0; x < colCount; x++)
+                {
+                    #><#= String.Format("double m{0}{1}", x + 1, y + 1) #><#
+
+                    if (x + 1 != colCount || y + 1 != rowCount)
+                    {
+                        #>, <#
+                    }
+                }
+
+                if (y + 1 < rowCount)
+                {
+                    #><#= Environment.NewLine #><#
+                    #>                         <#
+                }
+            }
+#>)
+        {
+<#
+            for (int y = 0; y < rowCount; y++)
+            {
+#>            <#
+                for (int x = 0; x < colCount; x++)
+                {
+                    #><#= String.Format("M{0}{1} = m{0}{1}; ", x + 1, y + 1) #><#
+                }
+                #><#= Environment.NewLine #><#
+            }
+#>
+        }
+
+        /// <summary>
+        /// Initialized a <#= structName #> with all values set to the same value
+        /// </summary>
+        /// <param name="value">The value to set all values to</param>
+        public <#= structName #>(double value)
+        {
+<#
+            for (int y = 0; y < rowCount; y++)
+            {
+#>            <#
+                for (int x = 0; x < colCount; x++)
+                {
+                    #><#= String.Format("M{0}{1} = ", x + 1, y + 1) #><#
+
+                    if (x + 1 == colCount && y + 1 == rowCount)
+                    {
+                        #>value;<#
+                    }
+                }
+                #><#= Environment.NewLine #><#
+            }
+#>
+        }
+
+<#
+            for (int y = 0; y < rowCount; y++)
+            {
+                for (int x = 0; x < colCount; x++)
+                {
+                    #>        <#
+                    #><#= String.Format("public double M{0}{1}", x + 1, y + 1) #>;<#
+                    #><#= Environment.NewLine #><#
+                }
+            }
+#>
+
+        public unsafe double this[int col, int row]
+        {
+            get
+            {
+                if (col < 0 || col >= ColumnCount)
+                    throw new ArgumentOutOfRangeException(nameof(col), $"Expected greater than or equal to 0 and less than {ColumnCount}, found {col}.");
+                if (row < 0 || row >= RowCount)
+                    throw new ArgumentOutOfRangeException(nameof(row), $"Expected greater than or equal to 0 and less than {RowCount}, found {row}.");
+
+                fixed (<#= structName #>* p = &this)
```

Comment:
Anyone know if this alignment, which works well on x86, will work on ARM and other platforms?